runtime/compiler/ilgen/Walker.cpp

```diff
@@ -6372,7 +6372,203 @@ TR_J9ByteCodeIlGenerator::loadFromCP(TR::DataType type, int32_t cpIndex)
+                     break;
+                  case 'S':
+                     autoboxClassSig = "Ljava/lang/Short;";
+                     autoboxClassSigLength = 17;
+                     typeClassBlock = comp()->fej9()->getClassFromSignature(autoboxClassSig, autoboxClassSigLength, method());
+                     valueOffset = comp()->fej9()->getObjectHeaderSizeInBytes()
+                                          + comp()->fej9()->getInstanceFieldOffset(typeClassBlock, "value", 5, "S", 1);
+                     break;
+                  case 'Z':
+                     autoboxClassSig = "Ljava/lang/Boolean;";
+                     autoboxClassSigLength = 19;
+                     typeClassBlock = comp()->fej9()->getClassFromSignature(autoboxClassSig, autoboxClassSigLength, method());
+                     valueOffset = comp()->fej9()->getObjectHeaderSizeInBytes()
+                                          + comp()->fej9()->getInstanceFieldOffset(typeClassBlock, "value", 5, "Z", 1);
+                     break;
+                  default:
+                     TR_ASSERT_FATAL(false, "Unrecognized primitive constant dynamic type signature");
+                     break;
+                  }
+               // Generate constant for resolved primitive condy.
+               if(!isCondyUnresolved)
+                  {
+                  TR::VMAccessCriticalSection primitiveCondyCriticalSection(comp()->fej9(),
+                                                            TR::VMAccessCriticalSection::tryToAcquireVMAccess,
+                                                            comp());
+                  if (primitiveCondyCriticalSection.hasVMAccess())
+                     {
+                     uintptrj_t* objLocation = (uintptrj_t*)_methodSymbol->getResolvedMethod()->dynamicConstant(cpIndex);
+                     uintptrj_t obj = *objLocation;
+                     TR_ASSERT(obj, "Resolved primitive Constant Dynamic-type CP entry %d must have autobox object", cpIndex);
+                     switch (returnTypeUtf8Data[0])
+                        {
+                        case 'I':
+                        case 'Z':
+                        case 'C':
+                        case 'S':
+                        case 'B':
+                           loadConstant(TR::iconst, *(int32_t*)(obj+valueOffset));
+                           break;
+                        case 'J':
+                           loadConstant(TR::lconst, *(int64_t*)(obj+valueOffset));
+                           break;
+                        case 'F':
+                           loadConstant(TR::fconst, *(float*)(obj+valueOffset));
+                           break;
+                        case 'D':
+                           loadConstant(TR::dconst, *(double*)(obj+valueOffset));
+                           break;
+                        default:
+                           break;
```

Comment:
need an assert here too to match the case above.

Reply:
Added assertion.